src/context/report-filter-context.tsx

```diff
@@ -0,0 +1,51 @@
+import { createContext, VNode, ComponentChildren } from "preact";
+import { route } from "../router";
+
+const DEFAULT_IAL = 1;
+const DEFAULT_ENV = "prod";
+
+interface ReportFilterContextValues {
+  start: Date;
+  finish: Date;
+  ial: 1 | 2;
+  agency?: string;
+  env: string;
+}
+
+/**
+ * preact-router does not parse "+" as spaces the way JS encodes them
+ */
+function pathWithParams(path: string, searchParams: URLSearchParams): string {
+  return `${path}?${searchParams.toString().replace(/\+/g, "%20")}`;
+}
+
+function defaultSetParameters(
+  params: Record<string, string>,
+  location: Location = window.location
+): void {
+  const searchParams = new URLSearchParams(location.search);
+  Object.keys(params).forEach((key) => searchParams.set(key, params[key]));
+  route(pathWithParams(location.pathname, searchParams));
+}
+
+const ReportFilterContext = createContext({
+  start: new Date(),
+  finish: new Date(),
+  ial: DEFAULT_IAL,
+  env: DEFAULT_ENV,
+  setParameters: defaultSetParameters,
+} as ReportFilterContextValues & { setParameters: (params: Record<string, string>) => void });
+
+function ReportFilterContextProvider({
+  children,
+  ...rest
+}: { children: ComponentChildren } & ReportFilterContextValues): VNode {
```

Comment:
Nit: Passing `setParameters` as a prop is allowed, but it doesn't do anything. If we don't want to make it configurable, we could use the [`Omit` utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys)?

```suggestion
}: { children: ComponentChildren } & Omit<ReportFilterContextValues, 'setParameters'>): VNode {
```

(Which could be extracted e.g. `type ReportFilterContextProviderProps = Omit<ReportFilterContextValues, 'setParameters'>`)

Else I'd expect something below like `{ ...rest, setParameters: setParameters ?? defaultSetParameters }`.

Reply:
Omit was what I was looking for: f6afa30c9fa9c3f8045dd8acc6637cd7a98e4ae9